src/main/java/com/spotinst/sdkjava/model/OceanGKELaunchSpecConverter.java

```diff
@@ -0,0 +1,450 @@
+
+            if (src.isSourceImageSet()) {
+                apiGroup.setSourceImage(src.getSourceImage());
+            }
+
+            if (src.isServiceAccountSet()) {
+                apiGroup.setServiceAccount(src.getServiceAccount());
+            }
+
+            if (src.isRootVolumeSizeSet()) {
+                apiGroup.setRootVolumeSize(src.getRootVolumeSize());
+            }
+
+            if (src.isRootVolumeTypeSet()) {
+                apiGroup.setRootVolumeType(src.getRootVolumeType());
+            }
+
+            if (src.isRestrictScaleDown()) {
+                apiGroup.setRestrictScaleDown(src.getRestrictScaleDown());
+            }
+            if (src.isMetadataSet()) {
+                apiGroup.setMetadata(metadataToDal(src.getMetadata()));
+            }
+            if (src.isInstanceTypesSet()) {
+                apiGroup.setInstanceTypes(src.getInstanceTypes());
+            }
+
+            if (src.isTaintsSet()) {
+                apiGroup.setTaints(oceanGKETaintsToDal(src.getTaints()));
+            }
+
+            if (src.isLabelsSet()) {
+                apiGroup.setLabels(labelsToDal(src.getLabels()));
+            }
+
+            if (src.isAutoScaleSet()) {
+                apiGroup.setAutoScale(autoScaleToDal(src.getAutoScale()));
+            }
+
+            if (src.isResourceLimitsSet()) {
+                apiGroup.setResourceLimits(restourceLimitToDal(src.getResourceLimits()));
+            }
+
+            if (src.isOceanGKEShieldedInstanceConfigSet()) {
+                apiGroup.setOceanGKEShieldedInstanceConfig(shieldedInstanceToDal(src.getOceanGKEShieldedInstanceConfig()));
+            }
+            if (src.isStrategySet()) {
+                apiGroup.setStrategy(startegyToDal(src.getStrategy()));
+            }
+            if (src.isStorageSet()) {
+                apiGroup.setStorage(storageToDal(src.getStorage()));
+            }
+
+        }
+
+        return apiGroup;
+    }
+
+    private static List<ApiOceanGKETaints> oceanGKETaintsToDal(List<OceanGKETaints> taints) {
+
+        List<ApiOceanGKETaints> taintsListToReturn = new ArrayList<>();
+        ArrayList<OceanGKETaints> taintsList =  new ArrayList<>(taints);
+
+        for(int i = 0; i < taintsList.size();i++){
+            ApiOceanGKETaints retVal = ApiOceanGKETaints.Builder.get().build();
+
+            if(taintsList.get(i).isKeySet()){
+                retVal.setKey(taintsList.get(i).getKey());
+            }
+            if(taintsList.get(i).isValueSet()){
+                retVal.setValue(taintsList.get(i).getValue());
+            }
+            if(taintsList.get(i).isEffectSet()){
+                retVal.setEffect(taintsList.get(i).getEffect());
+            }
+
+            taintsListToReturn.add(retVal);
+
+        }
+
+        return taintsListToReturn;
+    }
+
+    private static List<ApiOceanGKELabels> labelsToDal(List<OceanGKELabels> labels){
+        List<ApiOceanGKELabels> labelsListToReturn = new ArrayList<>();
```

Comment:
Confusing why two new lists are needed here? We are copying from x to y list, so only y need to be created newly.